src/main/java/picard/sam/SamErrorMetric/CollectSamErrorMetrics.java

```diff
@@ -328,21 +347,41 @@ private void writeMetricsFileForAggregator(final BaseErrorAggregation locusAggre
     }
 
     /**
-     * Advance the iterator until at or ahead of locus. if there's a variant at the locus return true, otherwise false.
+     * If there's a variant at the locus return true, otherwise false.
      * <p>
-     * HAS SIDE EFFECTS!!! draws from vcfIterator!!!
+     * HAS SIDE EFFECTS!!! Queries the vcfFileReader
      *
-     * @param vcfIterator        a {@link VariantContext} iterator to advance (assumed sorted)
-     * @param locus              a {@link Locus} at which to examine the variants
-     * @param sequenceDictionary a dictionary with which to compare the Locatable to the Locus...
+     * @param vcfFileReader      a {@link VCFFileReader} to query for the given locus
+     * @param locusInfo          a {@link SamLocusIterator.LocusInfo} at which to examine the variants
      * @return true if there's a variant over the locus, false otherwise.
      */
-    private static boolean advanceIteratorAndCheckLocus(final PeekableIterator<VariantContext> vcfIterator, final Locus locus, final SAMSequenceDictionary sequenceDictionary) {
-        while (vcfIterator.hasNext() && (vcfIterator.peek().isFiltered() ||
-                CompareVariantContextToLocus(sequenceDictionary, vcfIterator.peek(), locus) < 0)) {
-            vcfIterator.next();
+    private static boolean checkLocus(final VCFFileReader vcfFileReader, final SamLocusIterator.LocusInfo locusInfo) {
+
+        boolean overlaps = false;
+
+        if (locusInfo != null) {
```

Comment:
querying every locus is faster than iterating over the entire VCF? I'm very confused about this.

Reply:
In giving dbSNP to filter out known variants, by default this tool will iterate over the VCF and ignore where the reads are.

This may be fine for when you go over a whole exome, but if I have a read on chr22 I don't want to have to wait for it to get there.  Since our VCFs are indexed by default it is trivial to just jump to the next read or variant, whichever is closer.  This saves a lot of time.

I don't think it's reasonable to have to subset dbSNP every time I want to test a specific region of the genome.  If I had to do this, then virtually every time I call this tool I would be subsetting dbSNP again by the extents of the reads.